specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2023-02-01-preview/purviewcatalog.json

```diff
@@ -0,0 +1,8540 @@
+          },
+          {
+            "$ref": "#/parameters/offset"
+          },
+          {
+            "$ref": "#/parameters/sort"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. An array of glossary terms for the given glossary or an empty list.",
+            "schema": {
+              "type": "array",
+              "description": "An array of glossary terms for the given glossary or an empty list.",
+              "items": {
+                "$ref": "#/definitions/AtlasGlossaryTerm"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/atlas/v2/glossary/{glossaryGuid}/terms/headers": {
+      "get": {
+        "tags": [
+          "Glossary"
+        ],
+        "description": "Get term headers belonging to a specific glossary. Recommend using limit/offset to get pagination result.",
+        "operationId": "Glossary_ListGlossaryTermHeaders",
+        "x-ms-examples": {
+          "Glossary_ListGlossaryTermHeaders": {
+            "$ref": "./examples/Glossary_ListGlossaryTermHeaders.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/glossaryGuid"
+          },
+          {
+            "$ref": "#/parameters/limit"
+          },
+          {
+            "$ref": "#/parameters/offset"
+          },
+          {
+            "$ref": "#/parameters/sort"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. An array of glossary terms for the given glossary or an empty list.",
+            "schema": {
+              "type": "array",
+              "description": "An array of glossary terms for the given glossary or an empty list.",
+              "items": {
+                "$ref": "#/definitions/AtlasRelatedTermHeader"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/glossary/{glossaryGuid}/terms/import": {
+      "post": {
+        "tags": [
+          "Glossary"
+        ],
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "description": "Import Glossary Terms from local csv file. Note the csv file should be uploaded in formData with key='file'.",
+        "operationId": "Glossary_ImportGlossaryTermsViaCsv",
+        "x-ms-examples": {
+          "Glossary_ImportGlossaryTermsViaCsv": {
+            "$ref": "./examples/Glossary_GetImportCSVOperationStatus.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Content-Type"
+          },
```

Comment:
Thanks. We want to let users understand how to set the content-type when calling our REST APIs directly as we noticed that some customers did not set the correct content-type when using postman to call this API. So, we want to provide more information in the public document in the same way as the storage service.
Here is the document from Storage service listing the requests headers including content-length and content-type. https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob?tabs=azure-ad#request-headers-all-blob-types

https://github.com/Azure/azure-rest-api-specs/blob/624dbc769880e5676ae8bb20d3c82ebd1783c64a/specification/storage/data-plane/Microsoft.BlobStorage/preview/2021-12-02/blob.json#L11357

Shall we also mark the content-type as optional? Also, the current documentation does not support showing formData parameters. So, we are trying to find a way to let customers know how to correctly use this API. Here is the work item tracked by Reference team. https://dev.azure.com/ceapex/Engineering/_workitems/edit/766112

Reply:
In OpenAPI, the set of allowed content types is defined by the "consumes" field of the operation. In this case, only one content type is allowed, "multipart/form-data". To avoid the potential for ambiguous/conflicting API descriptions, OpenAPI [does not allow `Content-type` to be defined explicitly](https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#fixed-fields-10).
> If `in` is `"header"` and the `name` field is `"Accept"`, `"Content-Type"` or `"Authorization"`, the parameter definition SHALL be ignored.

Given that only one content-type is allowed, I think the service should simply make this the default when content-type is not specified and then users don't even need to think about it.

If you really think you need to document that the only content-type they can pass is "multipart/form-data" you could add a statement to that effect in the operation description.

Reply:
Understand. Content-type header is removed. We will use another way to document this information. Thanks